### ISSUE TITLE
[bug] 3.9.3-8: backdoor quest only works for host.

### ISSUE THREAD

**Issue by r-pufky** [project-fika/Fika-Server#76]
The quest [backdoor](https://escapefromtarkov.fandom.com/wiki/Back_Door) does not trigger completed for non-host players, even with requirements met; quest progression sharing turn on or off.

we were only able to trigger this quest completion by having each player who needed it completed hosting the raid themselves. Quest progression sharing toggled on or off did not fix the issue. Multiple SPT/fika-server/fika-client versions did not fix the issue.

we have only able to complete it on the host system; tested SPT 3.9.5, 3.9.6, 3.9.8; with corresponding Fika server/plugins.

The only additional information is that the non-host players do not see the light switched on, indicating a potential state sync issue with this quest.

**Comment by Lacyway** [comment]
What light switch are you referring to? Can you provide screenshots?

**Comment by Lacyway** [comment]
Also, provide some logs that I can scan through.

**Comment by Lacyway** [comment]
For context, I just tried this on a dedicated session as a client and it finished for me. So I'm going to need more information to figure this out.

**Comment by r-pufky** [comment]
I have kept complete archives of each eft-fika server build between upgrades, so I can send you the complete server instance if you want to look at that as well. All clients are confirmed to have always been on the same SPT/Fika versions.

# 3.9.5
## Versions
* EFT (Released): 0.14.9.6.31020
* EFT (SPT Downgrade): 0.14.9.30626 (released 27th June 2024)
  * https://hub.sp-tarkov.com/files/file/672-spt-installer
* SPT: 3.9.5 - 30626 - e4ba92b
  * https://spt-releases.modd.in/SPT-3.9.5-30626-e4ba92b.7z
* Fika: 0.9.8983.25381
  * https://github.com/project-fika/Fika-Plugin/releases/download/v0.9.8983.25381/Fika.Release.0.9.8983.25381.zip
* (server) Fika-Server: 2.2.7
* (server) SPT: 3.9.5
* (server) Node: 20.11.1

## Players
### 662de5d10000a51fe407c3f4
* usec
* back door completed
* hosting
* win 10
* local network connection to server

### 662de733000539ddde3e7e6c
* bear
* back door started
* client
* win 10
* remote network connection

## Server
* debian bookworm

### Server build steps
```bash
apt update && apt dist-upgrade
apt install sudo vim ssh git git-lfs
adduser {USER} sudo
adduser eft
sudo mkdir /opt/eft
sudo chown eft:eft /opt/eft
sudo su - eft
export SPT_TAG=3.9.5
export FIKA_TAG=v2.2.7
export NODE_VERSION=20.11.1

git clone https://github.com/nvm-sh/nvm ~/.nvm
. ~/.nvm/nvm.sh
source ~/.bashrc
nvm install $NODE_VERSION

git clone --depth 1 --branch $SPT_TAG https://dev.sp-tarkov.com/SPT/Server ~/srv
cd ~/srv/project
git checkout HEAD^
git lfs fetch --all
git lfs pull
sed -i '/setEncoding/d' ~/srv/project/src/Program.ts
npm install
npm run build:release -- --arch=x64 --platform=linux
mv build/* /opt/eft/
rm -rfv ~/srv

git clone --branch $FIKA_TAG https://github.com/project-fika/Fika-Server /opt/eft/user/mods/fika-server
cd /opt/eft/user/mods/fika-server
git checkout HEAD^
npm install

cd /opt/eft
nohup timeout --preserve-status 25s ./SPT.Server.exe >/dev/null 2>&1
sed -i 's/127.0.0.1/0.0.0.0/g' /opt/eft/SPT_Data/Server/configs/http.json
chown -R eft:eft .

vim /opt/eft/user/mods/fika-server/assets/configs/fika.jsonc
  {
    "client": {
      "useBtr": true,
      "friendlyFire": true,
      "dynamicVExfils": false,
      "allowFreeCam": false,
      "allowSpectateFreeCam": true,
      "allowItemSending": true,
      "blacklistedItems": [],
      "forceSaveOnDeath": false,
      "mods": {
        "required": [],
        "optional": []
      },
      "useInertia": true,
      "sharedQuestProgression": true
    },
    "server": {
      "giftedItemsLoseFIR": false,
      "launcherListAllProfiles": false,
      "sessionTimeout": 5,
      "showDevProfile": false,
      "showNonStandardProfile": false
    },
    "natPunchServer": {
      "enable": false,
      "port": 6790,
      "natIntroduceAmount": 1
    }
    "dedicated": {
      "profiles": {
        "amount": 0
      },
      "scripts": {
        "generate": true,
        "forceIp": ""
      }
    },
    "background": {
      "enable": true,
      "easteregg": false
    }
  }

sed -i 's/CanRequireOnRagfair\":\ false/CanRequireOnRagfair\":\ true/g' /opt/eft/SPT_Data/Server/database/templates/items.json
sed -i 's/CanSellOnRagfair\":\ false/CanSellOnRagfair\":\ true/g' /opt/eft/SPT_Data/Server/database/templates/items.json

vim /etc/systemd/system/eft.service
  [Unit]
  Description=Escape from Tarkov (Coop) SPT/FIKA Server.

  [Service]
  Type=exec
  WorkingDirectory=/opt/eft
  ExecStart=/opt/eft/SPT.Server.exe
  Restart=on-failure
  User=eft
  Group=eft

  [Install]
  WantedBy=default.target

systemctl daemon-reload
systemctl enable eft
systemctl start eft
```

# Log
[server-2024-08-17.log](https://github.com/user-attachments/files/16677439/server-2024-08-17.log)

Attached is the redacted completion log from the server side, where client `662de733000539ddde3e7e6c` hosted the raid, and completed the quest successfully. see line **9765**

# Setup
```text                                                  
  ┌───────────────┐            ┌───────────────┐  
  │client f4      │            │client 6c      │  
  │host           │            │remote network │  
  │back door done │            │back door start│  
  │local network  │            │               │  
  │               │            │               │  
  │               │            │               │  
  │               │            │               │  
  │               │            │               │  
  │               │            │               │  
  └───────────────┘            └───────────────┘  
                                                  
  ┌───────────────┐                               
  │server         │                               
  │local network  │                               
  │               │                               
  │               │                               
  │               │                               
  │               │                               
  │               │                               
  │               │                               
  │               │                               
  └───────────────┘                               
```                                                  
                                                  
We'll reset server state to 3.9.8 this week and reproduce again with same setup; and grab logs from both clients as well as the server.

**Comment by Lacyway** [comment]
I need the client's log file after a raid where it fails to complete. Quest progression is 100% client based and local, so are the door states/levers.

**Comment by Lacyway** [comment]
Let me know when you have updated and tried again. Thanks.

**Comment by r-pufky** [comment]
> Let me know when you have updated and tried again. Thanks.

no worries. will get this done shortly! Thanks for looking into it.

**Comment by r-pufky** [comment]
Successfully reproduced. Here's the logs for the server, client-host, and client-client.

[2024-08-20-backdoor-reproduction.tar.gz](https://github.com/user-attachments/files/16685659/2024-08-20-.tar.gz)

## Client
* EFT (Released): 0.14.9.6.31020
* EFT (SPT Downgrade): 0.14.9.30626 (released 27th June 2024)
  * https://hub.sp-tarkov.com/files/file/672-spt-installer
* SPT: 3.9.8 - 30626 - a0c79ac
  * https://spt-releases.modd.in/SPT-3.9.8-30626-a0c79ac.7z
* Fika: 0.9.8995.14684
  * https://github.com/project-fika/Fika-Plugin/releases/download/v0.9.8995.14684/Fika.Release.0.9.8995.14684.zip

## Server
* Fika-Server: 2.2.8
* SPT: 3.9.8
* Node: 20.11.1

**Comment by Lacyway** [comment]
Unfortunately these are the wrong log files.

I need `%AppData%\..\LocalLow\Battlestate Games\EscapeFromTarkov\Player.log` from the client.

**Comment by r-pufky** [comment]
[2024-08-20-client-appdata-logs.tar.gz](https://github.com/user-attachments/files/16695389/2024-08-20-client-.tar.gz)

Here's the logs for the host and the client from appdata. If you need cleaner logs we can clear and reproduce again tonight or tomorrow.

**Comment by Lacyway** [comment]
Everything looks fine in the log. Could you please provide me a video of how it looks for the client?
I just did it back to back 2 times, works every time so I'm very lost here.

**Comment by r-pufky** [comment]
Sure! We will get it setup and run with a video. Might be a day or two.

**Comment by r-pufky** [comment]
Thanks for waiting -- reproduced with videos.

Shared quest progression is **enabled**; same versions as above

[client host logs](https://github.com/user-attachments/files/16744542/client-host.tar.gz)
* quest already completed
* no quest interactions

[client passive logs](https://github.com/user-attachments/files/16744548/client-passive.tar.gz)|[client passive video](https://drive.google.com/file/d/1mzUOBibAb3w3sT5eSrgATGFSASqvcVs_/view?usp=drive_link)
* shows quest before connecting
* no quest interactions (level pull, D2 exit door activation)
* show quest after finishing with survive (not completed)

[client-active.tar.gz](https://github.com/user-attachments/files/16744559/client-active.tar.gz)|[client active video](https://drive.google.com/file/d/1i9VWzjfzwqZevjrSZtboAwRxgnfiFX95/view?usp=drive_link)
* shows quest before connecting
* all quest interactions (level pull, D2 exit door activation)
* show quest after finishing with survive (not completed)

**Comment by Lacyway** [comment]
Do you have auto extract on?

**Comment by r-pufky** [comment]
I believe everyone has auto-extract on.

**Comment by Lacyway** [comment]
Can you try without?

**Comment by r-pufky** [comment]
> Can you try without? Just enable free cam, no bots and do a quick run.

Will do. We will run it tomorrow, they have already gone to bed for the night.

**Comment by r-pufky** [comment]
We just re-ran the test. with **auto-extract** disabled on all the machines, the mission finished correctly for everyone (client-active, client-passive).

**Comment by Lacyway** [comment]
> We just re-ran the test. with **auto-extract** disabled on all the machines, the mission finished correctly for everyone (client-active, client-passive).

Sweet, I'll fix this for 3.10. Thanks!
